diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8b859a5..56b7ad7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -221,12 +221,20 @@ install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/
   PATTERN "internal" EXCLUDE
 )
 install(TARGETS freetype
+  EXPORT freetype-targets
   RUNTIME DESTINATION bin
   LIBRARY DESTINATION lib
   ARCHIVE DESTINATION lib
   FRAMEWORK DESTINATION Library/Frameworks
 )
 
+set(CONFIG_INSTALL_PATH "lib/cmake/freetype" CACHE STRING "location to install cmake config files")
+
+install(EXPORT freetype-targets
+  DESTINATION ${CONFIG_INSTALL_PATH}
+  FILE freetype-config.cmake
+)
+
 # Packaging
 # CPack version numbers for release tarball name.
 set(CPACK_PACKAGE_VERSION_MAJOR ${VERSION_MAJOR})
