diff --git a/CMakeLists.txt b/CMakeLists.txt
index 00c7e62..f400c44 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -105,6 +105,10 @@ set(PROJECT_VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH})
 # Compiler definitions for building the library
 add_definitions(-DFT2_BUILD_LIBRARY)
 
+if(MSVC)
+  add_definitions(-D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS)
+endif()
+
 # Specify library include directories
 include_directories("${PROJECT_SOURCE_DIR}/include")
 
diff --git a/include/freetype.h b/include/freetype.h
index 27fd44b..2798005 100644
--- a/include/freetype.h
+++ b/include/freetype.h
@@ -37,7 +37,9 @@
 
 FT_BEGIN_HEADER
 
-
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP)
+#define generic GenericFromFreeTypeLibrary
+#endif
 
   /*************************************************************************/
   /*                                                                       */
@@ -1739,6 +1741,10 @@ FT_BEGIN_HEADER
 
   } FT_GlyphSlotRec;
 
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP)
+#undef generic
+#endif
+
 
   /*************************************************************************/
   /*************************************************************************/
diff --git a/src/base/ftobjs.c b/src/base/ftobjs.c
index ee15a01..4c96009 100644
--- a/src/base/ftobjs.c
+++ b/src/base/ftobjs.c
@@ -449,6 +449,10 @@
 
   /* documentation is in ftobjs.h */
 
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP)
+#define generic GenericFromFreeTypeLibrary
+#endif
+
   FT_BASE_DEF( void )
   FT_Done_GlyphSlot( FT_GlyphSlot  slot )
   {
@@ -960,6 +964,10 @@
     FT_FREE( face );
   }
 
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP)
+#undef generic
+#endif
+
 
   static void
   Destroy_Driver( FT_Driver  driver )
